manifests/backup/mysqldump.pp

```diff
@@ -28,7 +30,7 @@
     ensure     => $ensure,
     user       => "${backupuser}@localhost",
     table      => '*.*',
-    privileges => [ 'SELECT', 'RELOAD', 'LOCK TABLES', 'SHOW VIEW', 'PROCESS' ],
+    privileges => [ 'SELECT', 'RELOAD', 'LOCK TABLES', 'SHOW VIEW', 'PROCESS', 'TRIGGER' ],
```

Comment:
I've added a branch for the 'TRIGGER' privilege, as well as a unit test to check it.

backing up routines does not appear to require privileges above the ones already granted.